pyanaconda/packaging/dnfpayload.py

```diff
@@ -797,23 +795,20 @@ def preInstall(self, packages=None, groups=None):
             self.requiredPackages += packages
         self.requiredGroups = groups
 
-        # Write the langpacks config
-        pyanaconda.iutil.mkdirChain(DNF_PLUGINCONF_DIR)
-        langs = [self.data.lang.lang] + self.data.lang.addsupport
-
-        # Start with the file in /etc, if one exists. Otherwise make an empty config
-        if os.path.exists(_DNF_TARGET_LANGPACK_CONF):
-            shutil.copy2(_DNF_TARGET_LANGPACK_CONF, _DNF_INSTALLER_LANGPACK_CONF)
-        else:
-            with open(_DNF_INSTALLER_LANGPACK_CONF, "w") as f:
-                f.write("[main]\n")
-
-        # langpacks.conf is an INI style config file, read it and
-        # add or change the enabled and langpack_locales entries without
-        # changing anything else.
-        keys=[("langpack_locales", "langpack_locales=" + ", ".join(langs)),
-              ("enabled", "enabled=1")]
-        simple_replace(_DNF_INSTALLER_LANGPACK_CONF, keys)
+        # get all available languages in repos
+        available_langpacks = self._base.sack.query().available() \
+            .filter(name__glob="langpacks-*")
+        alangs = [p.name.split('-')[1] for p in available_langpacks]
+
+        # add base langpacks into transaction
+        for lang in [self.data.lang.lang] + self.data.lang.addsupport:
+            loc = pyanaconda.localization.find_best_locale_match(lang, alangs)
+            if not loc:
+                log.warning(
+                    "Selected lang %s does not match any available langpack",
+                    lang)
```

Comment:
The string doesn't have to be on a separate line. We are not too strict about this in anaconda and it would just make the code better readable.